test/version_solver_test.dart

```diff
@@ -1978,6 +2102,94 @@ void features() {
       });
     });
   });
+
+  group("with overlapping dependencies", () {
+    test("can enable extra features", () async {
+      await servePackages((builder) {
+        builder.serve('foo', '1.0.0', pubspec: {
+          "dependencies": {'bar': '1.0.0'},
+          "features": {
+            "stuff": {
+              "default": false,
+              "dependencies": {
+                'bar': {
+                  'features': {'stuff': true}
+                }
+              }
+            }
+          }
+        });
+
+        builder.serve('bar', '1.0.0', pubspec: {
+          "features": {
+            "stuff": {
+              "default": false,
+              "dependencies": {'baz': '1.0.0'}
+            }
+          }
+        });
+
+        builder.serve('baz', '1.0.0');
+      });
+
+      await d.appDir({
+        'foo': {'version': '1.0.0'}
+      }).create();
+      await expectResolves(result: {'foo': '1.0.0', 'bar': '1.0.0'});
+
+      await d.appDir({
+        'foo': {
+          'version': '1.0.0',
+          'features': {'stuff': true}
+        }
+      }).create();
+      await expectResolves(
+          result: {'foo': '1.0.0', 'bar': '1.0.0', 'baz': '1.0.0'});
+    });
+
+    test("can't disable features", () async {
```

Comment:
How is this test trying to disable the feature? It looks like both foo's regular dependency on bar and the feature dependency on bar both turn on stuff. Should one be false?